src/core/router/transports/ssu/packet.h

```diff
@@ -755,8 +755,11 @@ class SSUPacketParser : private kovri::core::InputByteStream {
   std::unique_ptr<SSUSessionCreatedPacket> ParseSessionCreated();
 
   /// @brief Parses a session confirmed packet, without the header
+  /// @param header Packet header
+  /// @note header is required to compute padding
   /// TODO: Support multiple fragments?
-  std::unique_ptr<SSUSessionConfirmedPacket> ParseSessionConfirmed();
+  std::unique_ptr<SSUSessionConfirmedPacket> ParseSessionConfirmed(
+      const SSUHeader* header);
```

Comment:
I think we should consider using a member instead of an argument. Benefits include easier access to parsed header data (so we're not passing header data to *packet* parsing functions) and class design consistency (since we *do* do header parsing here after all).

Any thoughts?